elide-core/src/main/java/com/yahoo/elide/core/PersistentResource.java

```diff
@@ -1486,13 +1488,18 @@ protected void setValue(String fieldName, Object value) {
         }
 
         // Queue the @*Update triggers iff this is not a newly created object (otherwise we run @*Create)
+
+        ChangeSpec changeSpec = new ChangeSpec(this, fieldName, original, value);
         boolean isNewlyCreated = requestScope.getNewPersistentResources().contains(this);
-        requestScope.queueTriggers(this, fieldName, (isNewlyCreated) ? CRUDAction.CREATE : CRUDAction.UPDATE);
+        requestScope.queueTriggers(this, fieldName,
+                (isNewlyCreated) ? CRUDAction.CREATE : CRUDAction.UPDATE, Optional.of(changeSpec));
         requestScope.queueTriggers(this, (isNewlyCreated) ? CRUDAction.CREATE : CRUDAction.UPDATE);
         auditField(new ChangeSpec(this, fieldName, original, value));
     }
 
-    <A extends Annotation> void runTriggers(Class<A> annotationClass, String fieldName) {
+    <A extends Annotation> void runTriggers(Class<A> annotationClass,
+                                            String fieldName,
+                                            Optional<ChangeSpec> changeSpec) {
```

Comment:
Much as it pains me to say, might it be worth while to provide an overloaded `runTriggers(Class, String)` that passes `Optional.empty()` to the full method since most places don't have a `ChangeSpec` to pass?